modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -1448,9 +1449,17 @@ protected Object getAggregateValue(FeatureVisitor visitor, SimpleFeatureType fea
                     st.setFetchSize(fetchSize);
                     rs = st.executeQuery( sql );
                 }
-             
+                Class resultType = getMapping(rs.getMetaData().getColumnType(1));
+                Class outputType = att == null ? resultType : att.getType().getBinding();
+                final Converter converter = Converters.getConverter( resultType, outputType, null);
+                
                 while(rs.next()) {
                     Object value = rs.getObject(1);
+                    try {
+                        value = converter.convert(value, outputType);
```

Comment:
Wondering if this could cause regressions, like, today code dealing with dimensions happily takes a java.util.Date subclass, but with this conversion attempt we might fail to convert a custom java.util.Date subclass to the expected target type. Also, this method can freely return null (see the javadoc), and this is a problem too, as null will be returned when a non null value was returned from database. 
Finally, if this is the only point using Converters.getConverter, we can just drop the new API and use the existing Converters.convert